Library/Homebrew/version.rb

```diff
@@ -380,114 +380,115 @@ def self._parse(spec, detected_from_url:)
       spec.stem
     end
 
+    VERSION_REGEXES.each do |target, regex, replacements|
+      str = case target
+      when :stem
+        stem
+      when :full
+        spec_s
+      end
+      match = regex.match(str)
+      next if match.blank?
+
+      version = match.captures.first
+      return version if replacements.blank?
+
+      old_chars, new_chars = replacements.flatten
+      return version.tr(old_chars, new_chars)
+    end
+
+    nil
+  end
+  private_class_method :_parse
+
+  VERSION_REGEXES = [
```

Comment:
Almost wondering if this could/should be an array of hashes or openstruct or another object. Thoughts?

Reply:
Hmm maybe a `VersionMatcher` or `VersionParser` class? It could have a method that takes the `spec` argument in the `Version::_parse` function, and then perhaps a `UrlVersionParser` subclass would process the full `spec_s`, while a `StemVersionParser` subclass would process only the `stem` (and we could move the `stem` extraction logic from `Version::_parse` to this subclass)